spackbot/pr_add_labels.py

```diff
@@ -0,0 +1,168 @@
+# Copyright 2013-2021 Lawrence Livermore National Security, LLC and other
+# Spack Project Developers. See the top-level COPYRIGHT file for details.
+#
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+
+import logging
+import re
+
+from gidgethub import routing
+
+
+logger = logging.getLogger(__name__)
+router = routing.Router()
+
+
+#: ``label_patterns`` maps labels to patterns that tell us to apply the labels.
+#:
+#: Entries in the ``dict`` are of the form:
+#:
+#: ```python
+#: "label": {
+#:     "attr1": [r"regex1.1", r"regex1.2"],
+#:     "attr2": [r"regex2.1", r"regex2.2", r"regex2.3"],
+#:     "attr3": r"regex3.1",
+#:     ...
```

Comment:
I would indent this to the same level as the code, at first I thought this was the entire dict and not just an entry in the dict. Prob won't be an issue in the docs (are we making docs??) but a little difficult to read in comment form.

Reply:
I wonder if we actually need support for lists since regexes can express that

Reply:
makes it easier to parse for mere mortals.